### PR TITLE
Support queries of session-level phenotypic attributes

### LINKED ISSUE #249: update query template to work with new session-level phenotypic data model
- [x] create local db with https://github.com/neurobagel/neurobagel_examples/blob/main/data-upload/pheno-bids-output/example_synthetic_pheno-bids.jsonld for testing new functionality
- [x] update query model: split query parameter for min # sessions into
    - min # pheno sessions
    - min # imaging sessions
- note: we won't have an "any" param cause it gets too complicated (it would need to be exclusive from min_num_pheno_sessions and min_num_imaging_sessions)
- note 2: default for either parameter cannot be 1, or this gets undesirably constrained when user doesn't fill out either field
- [x] update query template
- What do we return for a subject when they match the criteria?
  - Since the session ID of their phenotypic + imaging sessions can be different, we should return both separately + info for both (b/c otherwise, if user only sets an imaging data criteria, the matching sessions have no phenotypic attributes attached which is not useful)
  - [x] Separate entries for each session
  - [x] Per session entry, a new "session_type" key
- [x] in subject-level response, split `num_sessions` into phenotypic + imaging ses count
- [x] create new data model for session entry in `subject_data` response field
- [x] update docstrings of functions, remove TODOs
- [x] update tests

Reference query examples:

1. 2+ imaging sessions; imaging modality = T1
2. 2+ pheno sessions; age > 60; assessment = MoCA
3. 2+ imaging sessions; age > 60; assessment = MoCA
4. 2+ imaging sessions; 2+ pheno sessions; age > 60; assessment = MoCA

### LINKED ISSUE #251: Extend default timeout of `post_query_to_graph()`
The current default value is 5 s for some reason, which is even less than the httpx default (10 s).

We should change this to 30 s.

### PR DESCRIPTION
**Pull request by alyssadai**
- Closes #249 
- Closes #251 

See also:
- #262 
- #263

Changes proposed in this pull request:

- Default timeout for POST request to graph extended
- `min_num_sessions` query parameter split up for phenotypic and imaging sessions
- New data model created for a specific session result in the API response
  - Includes new key `"session_type"` to differentiate phenotypic from imaging data
- Removed unnecessary manual ordering of variables returned from graph
- SPARQL query template updated to query for session-level attributes and count matching phenotypic sessions based on session-level filters
- Existing tests updated

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [x] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: app/api/utility.py [hunk 1]
```diff
@@ -133,61 +136,74 @@ def create_query(
...
+    if min_num_phenotypic_sessions is not None:
+        subject_level_filters += (
+            "\n"
+            + f"FILTER (?num_phenotypic_sessions >= {min_num_phenotypic_sessions})."
+        )
+    if min_num_imaging_sessions is not None:
+        subject_level_filters += (
+            "\n"
+            + f"FILTER (?num_imaging_sessions >= {min_num_imaging_sessions})."
+        )
```
Comment (surchs) [inline app/api/utility.py:186]: should we move this section above pheno_filters now so the flow is more in line with the hierarchy (e.g. subject > pheno session - imaging session)?

File: app/api/utility.py [hunk 2]
```diff
@@ -196,29 +212,55 @@ def create_query(
...
+                        }}
+                        OPTIONAL {{
+                            ?phenotypic_session nb:isSubjectGroup ?subject_group.
+                        }}
+                        OPTIONAL {{
+                            ?phenotypic_session nb:hasAssessment ?assessment.
+                        }}
+                    }}
+                    {phenotypic_session_level_filters}
+                }} GROUP BY ?subject
```
Comment (surchs) [inline app/api/utility.py:253]: I think this is cleaner to wrap my head around than sticking both pheno and imaging sub-searches in the same `SELECT` statement - but just for my understanding: is that the only reason you made them into two separate sub-queries? Or is there some other aspect that makes this necessary and that I am missing?

**Review by surchs — APPROVED** [review]
Thanks @alyssadai, that's a pretty big PR. From reading through it all (most) makes a lot of sense to me and I added questions / suggestions where not. Take a look, but I think this is all good to go :cook:

File: tests/test_query.py [hunk 3]
```diff
@@ -305,7 +305,8 @@ async def mock_get(
         sex,
         diagnosis,
         is_control,
-        min_num_sessions,
+        min_num_imaging_sessions,
+        min_num_phenotypic_sessions,
         assessment,
         image_modal,
     ):
```
[the remark this answers is not in the document]
Comment (alyssadai) [reply tests/test_query.py:312]: Good idea. Tracked in https://github.com/neurobagel/api/issues/266

File: app/api/utility.py (hunk 2 above)
Comment (alyssadai) [reply app/api/utility.py:253]: Yes, I thought that two sub-queries (and sets of query filters) was more intuitive and easier to read.

From what I can tell, you can achieve functionally the same behaviour with a single `SELECT` statement!

**Pull request merged**